src/numerics/timestepping/stepper.py

```diff
@@ -277,15 +277,18 @@ def take_time_step(self, solver):
 		Time = solver.time
 		for istage in range(self.nstages):
 			dt = self.dt
-			solver.time = Time + dt
+			#HACK -> Need to confirm that this is the right thing to do!
+
```

Comment:
In main we have the solver.time attribute updated prior to the get_residual and mult_inv_mass_matrix function calls. This becomes an issue for time dependent source terms. We were previously evaluating any source term at the next time step rather than the current one. This never showed up before because we rarely have time dependent source terms.